reciprocalspaceship/dataset.py

```diff
@@ -1287,31 +1287,83 @@ def canonicalize_phases(self, inplace=False):
 
         return self
 
-    def to_reciprocalgrid(self, key, gridsize):
+    def get_gridsize(self, dmin=None, sample_rate=3.0):
+        """
+        Determine an appropriate 3D grid size for reflection data.
+
+        Returns the smallest grid size that yields a real-space grid spacing
+        of at most `dmin`/`sample_rate` (in Å). The returned grid size will
+        be 'FFT-friendly' (2, 3, or 5 are the largest prime factors), and will
+        obey any symmetry constraints of the spacegroup.
+
+        Parameters
+        ----------
+        dmin : float
+            Highest-resolution reflection to consider for grid size
+        sample_rate : float
+            Sets the minimal grid spacing relative to dmin. For example,
+            `sample_rate=3` corresponds to a real-space sampling of dmin/3.
+            Value must be >= 1.0 (default: 3.0)
+
+        Returns
+        -------
+        list(int, int, int)
+            Grid size with desired spacing (list of 3 integers)
+        """
+        if dmin is None:
+            dmin = self.compute_dHKL().dHKL.min()
+
+        return utils.get_gridsize(
+            self.cell, dmin=dmin, sample_rate=sample_rate, spacegroup=self.spacegroup
+        )
+
+    def to_reciprocalgrid(self, key, sample_rate=3.0, dmin=None, gridsize=None):
         """
         Set up reciprocal grid with values from column, ``key``, indexed by
-        Miller indices. A 3D numpy array will be initialized with the
-        specified gridsize. Any missing Miller indices are initialized to
-        zero.
+        Miller indices.
 
         Notes
         -----
         - The data being arranged on a reciprocal grid must be compatible
           with a numpy datatype.
+        - Any missing Miller indices are initialized to zero.
+        - The grid size determined using `sample_rate` and `dmin` will depend
+          on the cell parameters of the dataset. If the grid size must be
+          consistent across different isomorphous cell parameters, `gridsize`
+          can be explicitly provided.
 
         Parameters
         ----------
         key : str
             Column label for value to arrange on reciprocal grid
+        sample_rate : float
+            Sets the minimal grid spacing relative to dmin. For example,
+            `sample_rate=3` corresponds to a real-space sampling of dmin/3.
+            (default: 3.0)
+        dmin : float
+            Highest-resolution reflection to consider for grid size. If None,
+            `dmin` will be set to the highest resolution reflection in the
+            dataset (default: None)
```

Comment:
Need to say: "The reflections used to populate the grid will also be truncated to resolution `>=dmin`. "

Reply:
updated docstring